Sources/ScreenObject/ScreenObject.swift

```diff
@@ -64,13 +73,26 @@ open class ScreenObject {
         self.app = app
         self.expectedElementGetters = [expectedElementGetter]
         self.waitTimeout = waitTimeout
-        try waitForScreen()
+        try waitForScreen(firstElementOnly: true)
     }
 
+    /// Waits for the first element from expectedElementGetters to load (if `firstElementOnly` is `true`)
+    /// or, by default, for all elements to load (`firstElementOnly` is `false`).
     @discardableResult
-    func waitForScreen() throws -> Self {
-        try XCTContext.runActivity(named: "Confirm screen \(self) is loaded") { (activity) in
-            try expectedElementGetters.forEach { getter in
+    public func waitForScreen(firstElementOnly: Bool = false) throws -> Self {
+        var testedGetters: Array<(XCUIApplication) -> XCUIElement>,
+            activityDescription: String
+
+        if firstElementOnly {
+            testedGetters = [expectedElementGetters.first!]
```

Comment:
Using force unwrap because the array is already checked for non-emptiness in `init`.